### DISCUSSION TITLE
"Memoize property assignments" treating array as always empty?

### DISCUSSION THREAD

**Discussion by AlbinoDrought** [vimeo/psalm#10069] [Q&A]
https://psalm.dev/r/f10e0af67c

Is there something I am doing wrong in the above example, or do I just need to disable this setting?

Cheers

**Comment by orklah** [comment]
It's not something you're doing  wrong per se. You may disable this setting but you'll probably discover that you also depend heavily on it in other places

There's just no perfect solution here. What I try to do in those cases is to make the emptyness check (or any other check) also encapsulated in a method, that way Psalm can't see that the variable is empty and then complain
